### DISCUSSION TITLE
Snippets in Assistant Chat

### DISCUSSION THREAD

**Discussion by FalkZ** [zed-industries/zed#22431] [Feature Requests]
Hi there,

First of all, thank you for creating such a wonderful editor! :)

# Problem Statement

I would like to use snippets within the assistant chat, as this would open up numerous opportunities, for example:
```json
{
   // adding frequently used links
   "body": ["/fetch https://github.com/raycast/script-commands#api"],
   // or prompts
   "body": ["Improve this text but keep the style as similar as possible."],
}
```

# Solution Proposal

Create a snippet "namespace" for assistant snippets, such as the file `assistant.json`, and utilize the same completion (or part of it) as in the editor itself.

**Alternative:** I could also create an extension that utilizes the slash command `/snippet [snippet name]` for this purpose ([Reference](https://zed.dev/docs/extensions/slash-commands)). Although I think it would be nicer to be integrated into Zed.

---
I would be happy to contribute this feature if you're interested. (I could probably use some guidance on where to start.)

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
